test-harness/src/test/groovy/com/netflix/conductor/test/integration/SubWorkflowSpec.groovy

```diff
@@ -443,6 +443,17 @@ class SubWorkflowSpec extends AbstractSpecification {
             tasks[2].status == Task.Status.SCHEDULED
         }
 
+        and: "the workflow is in a RUNNING state"
+        workflowSweeper.sweep([workflowInstanceId], workflowExecutor, workflowRepairService)
```

Comment:
Is this sweep needed? Since retry is invoked on the parent, it will be put to RUNNING state; and its subworkflow task will be set to IN_PROGRESS state by the subworkflow.

Reply:
Yes. In L431, `retry` is called with `resumeSubWorkflows=true`, which works like this,

1. Finds the failed sub workflow and retries it.
2. Recursively sets the SUB_WORKFLOW task of the parents to IN_PROGRESS state.

At the end of the method, the failed sub workflow in the tree is in RUNNING state and all its parents are still in FAILED state, but the SUB_WORKFLOW tasks are in IN_PROGRESS. A sweep will sync the state of the workflow with the state of its task.